docs(icons): icons doc page fix

ju-Skinner opened the pull request:
# Description

An issue was discovered with the Icons doc page.  

In the previous version of Storybook `Canvas`, was a way to put things in a a container. The current version seems to have a bug, as it populates with the Story default story if no attributes are set.

To fix it, I swapped it out with our `DocCanvas` component.

|before|after|
|--|--|
|<img width="1062" alt="image" src="https://github.com/user-attachments/assets/244cb180-78e5-495e-9861-7f53f6c9e986">|<img width="1059" alt="image" src="https://github.com/user-attachments/assets/54bfcfa7-8e77-48c6-bd47-08271020933e">|



## Type of change

- [X] This change requires a documentation update

# How Has This Been Tested?

- [X] other: Storybook

# Checklist:

If not applicable, leave options unchecked.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

AndrwM commented:
@pixelflips Yep! That's handled in a follow up PR here #316 to keep this diff clean

pixelflips commented:
> @AndrwM I just pulled down this branch and it seems like the panels are missing padding.
> 
> May not be specifically related to this branch, but I ran bundle, install, and relaunched my server and still seeing it. Just let me know.

@AndrwM nevermind I just saw your note about this in the description.

philschanely commented on app/views/examples/objects/sortable/_markup.html.erb:
```diff
@@ -0,0 +1,25 @@
+
+  <section class="sage-sortable__item" data-js-sortable-update-url="/update/url/here">
+    <div class="sage-sortable__item-content">
+      <h1 class="t-sage-body">Batman</h1>
```
If these are type elements that will typically be a part of this component I wouldn't argue with making them elements of the block such as `sage-sortable__item-title` and sage-sortable__item-tagline` and moving the type spec setting therein.

AndrwM replied:
@croswell and I talked about this a bit. – My preference would be to stick with using these type helpers because I'm thinking in the future we can use different types of elements here (small headings, labels, etc). Using `sage-sortable__item-title` seemed like a pre-emptive optimization.

This is a pattern that is used else where for Product sorting so I wanted to keep it as flexible as possible.

philschanely commented on lib/sage-frontend/javascript/system/init.js:
```diff
@@ -1,8 +1,12 @@
 Sage.init = function(elementNamesToInit) {
-  var shouldInit = function(elementName, selector) {
+  let shouldInit = function(elementName, selector) {
```
If we're OK moving to `let` etc. can we now safely also use arrow functions and imports too? Just curious.

AndrwM replied:
You bet we can! 🎉 🎉 🎉

philschanely replied:
![](https://media.giphy.com/media/xTiN0CNHgoRf1Ha7CM/giphy-downsized.gif)

monkeypox8 replied:
haha! that GIF is amazing

philschanely commented on lib/sage-frontend/javascript/system/sortable.js:
```diff
@@ -0,0 +1,50 @@
+  const SELECTOR_CONTAINER = 'data-js-sortable';
+  const SELECTOR_ITEM_UPDATE_URL = 'data-js-sortable-update-url';
+  const SETTINGS = {
+    ghostClass: 'sage-sortable__item--ghost',
```
👻

philschanely commented on lib/sage-frontend/stylesheets/system/core/_variables.scss:
```diff
@@ -85,7 +85,7 @@ $sage-btn-shadow-base: sage-shadow(sm) !default;
 
 // Focus outline
-$sage-btn-focus-outline-spacing: sage-spacing(2xs) !default;
+$sage-btn-focus-outline-spacing: ( sage-spacing(2xs) / 2) !default;
```
Not just `rem(2px)`? :-P that or perhaps add `3xs`?

AndrwM replied:
`3xs` is a great idea, thanks!

philschanely commented on lib/sage-frontend/stylesheets/system/patterns/objects/_dropdown.scss:
```diff
@@ -41,8 +41,7 @@ $-dropdown-item-focus-line-spacing-x: sage-spacing(sm);
   position: absolute;
-  left: 0;
-  top: calc(100% + #{sage-spacing(xs)});
+  top: calc(100% - #{( sage-spacing(xs) / 2)});
```
Are you doing this math particularly to remind us that we are halving the `xs` value rather than just using `2xs`?

AndrwM replied:
Haha, whoops – thanks for catching

AndrwM commented on lib/sage-frontend/stylesheets/system/patterns/objects/_sortable.scss:
```diff
@@ -0,0 +1,73 @@
+  overflow: hidden;
+
+  & > * {
+    user-select: none;
```
Yep, a stylistic decision to prevent and accidental text selection on drag.
I'll remove 👍

AndrwM commented on lib/sage-frontend/stylesheets/system/patterns/objects/_panel.scss:
```diff
@@ -7,27 +7,35 @@
+  min-height: rem(60px);
+}
+
+.sage-panel__header-right {
```
This is a child of `.sage-panel__header`. – Just for right-aligned content in the header (buttons, label, small-text, etc).

Example I have in mind:
```
.sage-panel
  .sage-panel__header
    .t-heading-2
    .t-body
    .sage-panel__header-right // Elements in this container are "floated" next to each other
        .sage-btn
        .sage-label

```

I don't think I'm _all the way there_ as I haven't tested the configuration above what that's what I had in mind in terms of structure.

monkeypox8 replied:
Ah, I see. Disregard!

AndrwM commented on lib/sage-frontend/stylesheets/system/patterns/objects/_panel.scss:
```diff
@@ -7,27 +7,35 @@
+}
+
+.sage-panel__header-right {
+  margin: 0 0 0 auto;
```
See above comment. This is a child of `.sage-panel__header`.

AndrwM commented on lib/sage-frontend/javascript/system/util.js:
```diff
@@ -19,11 +19,29 @@ Sage.util = (function(Sage) {
+
+    xhr.onload = function (evt) {
+      let elScript = document.createElement('script');
+      elScript.setAttribute('type', 'text/javascript')
```
Ty!

AndrwM commented:
Fixed the issues found above. Landing. Thank you everyone!

ju-Skinner committed:
docs(icons): swap Storybook Canvas component to DocCanvas

with the recent update Storybook changed the behavior of their Canvas component

ju-Skinner committed:
docs(icons): change the icons in order to reduce any confusion

ju-Skinner committed:
chore: remove function that is not longer needed

monicawheeler commented on libs/core/src/components/pds-icon/docs/pds-icon.mdx:
```diff
@@ -11,11 +11,11 @@ categories: Actions and Features. In order to ensure that the icons are adding t
 <DocCanvas client:only mdxSource={{
-  react: `<PdsIcon name="upload"></PdsIcon>`,
-  webComponent: `<pds-icon name="upload"></pds-icon>`
+  react: `<PdsIcon name="youtube"></PdsIcon>`,
```
I’m on mobile so I can’t check this visually so I’m curious if we want to avoid a branded icon example here? If so, maybe instead of YouTube maybe using the Kajabi icon or some other generic one?

ju-Skinner replied:
Took your advice and changed to `thumb-up-filled`

monicawheeler reviewed (approved):
Non blocking comment
